include/lsst/afw/geom/SpanSet.h

```diff
@@ -48,6 +63,63 @@ namespace lsst { namespace afw { namespace geom {
  */
 enum class Stencil { CIRCLE, BOX, MANHATTAN };
 
+// Forward declaration of the SpanSet class
+class SpanSet;
+
+/** @brief Create a SpanSet from a mask.
+ *
+ * Create a SpanSet from a class. The default behaivor is to include any pixels which have any
+ * bits set. More complex selection/filtering of bit patterns can be done by supplying a comparator
+ * function.
+ *
+ * @param mask - mask to convert to a SpanSet
+ * @param comparator - Functor object to use in the decision to include pixel in SpanSet. Should return
+ *                     true when a given pixel in the mask should be part of the SpanSet, and false
+ *                     otherwise. The functor takes a single value taken from the mask at the
+ *                     pixel under consideration. Defaults to evaluating true if the mask has bits set,
+ *                     and false otherwise.
+ *
+ * @tparam T - Pixel type of the Mask
+ * @tparam F - Type of the functor
+ */
+template <typename T, typename F = details::MaskToSpanSetDefaultFunctor<T>>
+std::shared_ptr<geom::SpanSet> maskToSpanSet(image::Mask<T> const & mask,
+                               F comparator=details::MaskToSpanSetDefaultFunctor<T>()) {
+    std::vector<Span> tempVec;
+    std::size_t startValue{0};
+    bool started{false};
+    auto const & maskArray = mask.getArray();
+    auto dimensions = maskArray.getShape();
+    for (size_t y{0}; y < dimensions[0]; ++y) {
```

Comment:
See also [RFC-242](https://jira.lsstcorp.org/browse/RFC-242) (which is unresolved, but I think this case is an argument in favor of my proposal to prefer C++98 syntax for initialization).